kolibri/plugins/management/assets/src/state/actions.js

```diff
@@ -420,18 +419,20 @@ function updateUser(store, stateUser) {
     // then assign the new role
     roleAssigned = new Promise((resolve, reject) => {
       // Take care of previous roles if necessary (will autoresolve if not)
-      handlePreviousRoles.catch(error => reject(error));
```

Comment:
In Admin -> Learner, this would delete the previous roles. I caught for errors, but the state never saw these updates because I didn't wait for the promise to return.

Might have worked if I just popped items off of the array, rather than assigning it to an empty one.